packages/apps/reputation-oracle/server/src/modules/qualification/qualification.service.spec.ts

```diff
@@ -1,348 +1,337 @@
-        .fn()
-        .mockRejectedValueOnce(new Error());
+      mockQualificationRepository.findByReference.mockResolvedValueOnce(
+        qualificationEntity as QualificationEntity,
+      );
+      mockUserRepository.findWorkersByAddresses.mockResolvedValueOnce([user]);
 
-      await expect(
-        qualificationService.createQualification(createQualificationDto),
-      ).rejects.toThrow(Error);
-    });
-  });
+      const result = await service.assign(reference, [
+        user.evmAddress as string,
+      ]);
 
-  describe('getQualifications', () => {
-    it('should return a list of qualifications', async () => {
-      const qualifications = [
-        {
-          reference: 'ref1',
-          title: 'title1',
-          description: 'desc1',
-        },
-      ];
-      qualificationRepository.getQualifications = jest
-        .fn()
-        .mockResolvedValue(qualifications);
+      expect(result).toMatchObject({
+        success: [user.evmAddress],
+        failed: [],
+      });
+    });
 
-      const result = await qualificationService.getQualifications();
+    it('should fail to assign user not in active status', async () => {
+      const reference = faker.string.uuid();
+      const qualificationEntity = {
+        reference,
+        title: faker.string.alpha(),
+        description: faker.string.alpha(),
+      };
+      const user = generateWorkerUser({
+        privateKey: generateEthWallet().privateKey,
+        status: UserStatus.INACTIVE,
+      });
 
-      expect(result).toEqual(qualifications);
+      mockQualificationRepository.findByReference.mockResolvedValueOnce(
+        qualificationEntity as QualificationEntity,
+      );
+      mockUserRepository.findWorkersByAddresses.mockResolvedValueOnce([user]);
+
+      const result = await service.assign(reference, [
+        user.evmAddress as string,
+      ]);
+
+      expect(result).toMatchObject({
+        success: [],
+        failed: [
+          {
+            evmAddress: user.evmAddress as string,
+            reason: 'User is not in active status',
+          },
+        ],
+      });
     });
 
-    it('should return a list of qualifications with null expiresAt', async () => {
-      const qualifications = [
-        {
-          reference: 'ref1',
-          title: 'title1',
-          description: 'desc1',
-        },
-      ];
-      qualificationRepository.getQualifications = jest
-        .fn()
-        .mockResolvedValue(qualifications);
+    it('should throw NOT_FOUND error', async () => {
+      const reference = faker.string.uuid();
+      const user = generateWorkerUser({
+        privateKey: generateEthWallet().privateKey,
+      });
 
-      const result = await qualificationService.getQualifications();
+      mockQualificationRepository.findByReference.mockResolvedValueOnce(null);
 
-      expect(result).toEqual(qualifications);
+      await expect(
+        service.assign(reference, [user.evmAddress as string]),
```

Comment:
No need to generate user here, you can use just `faker.finance.ethereumAddress()`